source/Cosmos.HAL2/BlockDevice/IDE.cs

```diff
@@ -0,0 +1,79 @@
+using System;
+using Cosmos.HAL.BlockDevice;
+
+namespace Cosmos.HAL.BlockDevice
+{
+    public class IDE
+    {
+        private static PCIDevice xDevice = HAL.PCI.GetDeviceClass(0x01, 0x01);
```

Comment:
Can you please use enums instead of hardcoded values? You should create two enums one for the possible ClassesId and another for SubClassesId, look here for the valid values: http://wiki.osdev.org/PCI#Class%20Codes

Reply:
Well then, i guess i'm gonna change the entire ids to enums, from parameters to arguments